### PR TITLE
samples: timer

### PR DESCRIPTION
**Pull request by wilu-nordic**
Only the last commit should be checked.

### CODE REVIEW

File: samples/timer/README.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+.. table-from-sample-yaml::
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the TIMER driver to confgure and run timer.
+The TIMER is configured to generate an interrupt after ~400 ms.
```
Comment (annakielar) [inline samples/timer/README.rst:25]:
```suggestion
The Timer driver is configured to generate an interrupt after ~400 ms.
```
**The Timer driver** or **The timer** (any)?

File: samples/timer/README.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+If the interrupt from the PMIC does not occur within 500 ms, an error message is going to be printed on the console.
```
Comment (annakielar) [inline samples/timer/README.rst:26]:
```suggestion
If the interrupt from the PMIC does not occur within 500 ms, an error message is printed on the console.
```

File: samples/timer/README.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+.. include:: /includes/build_and_run.txt
+
+Testing
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+
```
Comment (annakielar) [inline samples/timer/README.rst:87]: Any Timer-specific steps for testing?

File: samples/timer/src/main.c [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <npmx_core.h>
+#include <npmx_driver.h>
+#include <npmx_timer.h>
+
+#define LOG_MODULE_NAME timer
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+/* Timeout in milliseconds within npmx device should call a timer interrupt. */
```
Comment (annakielar) [inline samples/timer/src/main.c:16]:
```suggestion
/* Timeout in milliseconds during which an npmx device should call a timer interrupt. */
```

File: samples/timer/src/main.c [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	/* Register SHIPHOLD callback. */
+	npmx_core_register_cb(npmx_instance, timer_callback, NPMX_CALLBACK_TYPE_EVENT_SHIPHOLD);
+
+	/* Timer general-purpose mode use SHIPHOLD WATCHDOG bit to indicate interrupt. */
+	npmx_core_event_interrupt_enable(npmx_instance, NPMX_EVENT_GROUP_SHIPHOLD,
+					 NPMX_EVENT_GROUP_SHIPHOLD_WATCHDOG_MASK);
+
+	/* Fill timer's configuration structure to work with general-purpose mode. */
+	npmx_timer_config_t timer_config = {
+		.mode = NPMX_TIMER_MODE_GENERAL_PURPOSE, /* General purpose mode. */
```
Comment (annakielar) [inline samples/timer/src/main.c:84]:
```suggestion
		.mode = NPMX_TIMER_MODE_GENERAL_PURPOSE, /* General purpose timer mode. */
```

**Review by annakielar — CHANGES_REQUESTED** [review]

File: samples/timer/README.rst (hunk 1 above)
Comment (wilu-nordic) [reply samples/timer/README.rst:25]: Periphery

File: samples/timer/src/main.c [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+{
+	LOG_ERR("Timeout, interrupt not received within %d ms.", APP_TIMEOUT_MILLISECONDS);
+}
+
+/* Variable filled in @ref main() when timer starts working. */
+static volatile uint32_t start_time;
+
+/**
+ * @brief Callback function to be used when SHIPHOLD WATCHDOG event occurs.
+ *        SHIPHOLD WATCHDOG is used by the timer in general-purpose mode.
```
[the remark this answers is not in the document]
Comment (wilu-nordic) [reply samples/timer/src/main.c:37]: I think can be just `the general purpose timer`.

**Review by annakielar — APPROVED** [review]

**Review by annakielar — APPROVED** [review]

**Review by annakielar — APPROVED** [review]

File: samples/timer/src/main.c [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+static void timer_callback(npmx_instance_t *p_pm, npmx_callback_type_t type, uint8_t mask)
+{
+	/* Cancel timeout. */
+	k_work_cancel_delayable(&timeout_timer);
+
+	/* Calculate elapsed time and print log info. */
+	LOG_INF("Elapsed time: %d ms.", k_uptime_get_32() - start_time);
+
+	/* Call generic callback to print interrupt data. */
+	p_pm->generic_cb(p_pm, type, mask);
```
Comment (masz-nordic) [inline samples/timer/src/main.c:52]: Not necessary here?

Comment (wilu-nordic) [reply samples/timer/src/main.c:52]: It prints:
```
[00:00:00.856,781] <dbg> NPMX: generic_callback: SHIPHOLD:
[00:00:00.856,811] <dbg> NPMX: generic_callback:        Watchdog Timeout Warn
```

In my opinion, it is worth having it here.

File: samples/timer/README.rst [hunk 8]
```diff
@@ -0,0 +1,91 @@
+.. _timer_sample:
+
+TIMER
```
Comment (annakielar) [inline samples/timer/README.rst:3]:
```suggestion
Timer
```

File: samples/timer/README.rst [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+.. table-from-sample-yaml::
+
+The sample also requires an nPM1300 EK.
+
+Overview
+********
+
+This sample uses the Timer driver to configure and run a timer.
+The TIMER peripheral is configured to generate an interrupt after ~400 ms.
+If the interrupt from the PMIC does not occur within 500 ms, an error message is printed on the console.
```
Comment (annakielar) [inline samples/timer/README.rst:26]:
```suggestion
If the interrupt from the PMIC does not occur within 500 ms, an error message is printed to the console.
```

**Review by annakielar — CHANGES_REQUESTED** [review]

**Review by annakielar — APPROVED** [review]

**Review by masz-nordic — APPROVED** [review]

**Pull request merged**